test/functional/test_framework/test_node.py

```diff
@@ -469,6 +469,12 @@ def blocks_path(self) -> Path:
     def blocks_key_path(self) -> Path:
         return self.blocks_path / "xor.dat"
 
+    @property
+    def read_xor_key(self) -> bytes:
+        with open(self.blocks_key_path, "rb") as xor_f:
+            NUM_XOR_BYTES = 8 # From InitBlocksdirXorKey::xor_key.size()
+            return xor_f.read(NUM_XOR_BYTES)
```

Comment:
Not sure if there are guidelines for this (and if yes, how strict we are), but I would tend to avoid using the "property" decorator for non-trivial methods that access state outside of the class (in this case, by interacting with the file system). At least I wouldn't expect as user that accessing a class field ever involves any I/O. I think dropping the property and keeping the name as-is "read_xor_key" might be just fine?

Reply:
Updated